specs/deneb/validator.md

```diff
@@ -79,29 +78,47 @@ def validate_blobs_and_kzg_commitments(execution_payload: ExecutionPayload,
 
 3. If valid, set `block.body.blob_kzg_commitments = blob_kzg_commitments`.
 
-#### Constructing the `SignedBeaconBlockAndBlobsSidecar`
-To construct a `SignedBeaconBlockAndBlobsSidecar`, a `signed_beacon_block_and_blobs_sidecar` is defined with the necessary context for block and sidecar proposal.
+#### Constructing the `SignedBlobSidecar`s
 
-##### Block
-Set `signed_beacon_block_and_blobs_sidecar.beacon_block = block` where `block` is obtained above.
+To construct a `SignedBlobSidecar`, a `signed_blob_sidecar` is defined with the necessary context for block and sidecar proposal.
 
 ##### Sidecar
-Coupled with block, the corresponding blobs are packaged into a sidecar object for distribution to the network.
 
-Set `signed_beacon_block_and_blobs_sidecar.blobs_sidecar = sidecar` where `sidecar` is obtained from:
+Blobs associated with a block are packaged into sidecar objects for distribution to the network.
+
+Each `sidecar` is obtained from:
 ```python
-def get_blobs_sidecar(block: BeaconBlock, blobs: Sequence[Blob]) -> BlobsSidecar:
-    return BlobsSidecar(
-        beacon_block_root=hash_tree_root(block),
-        beacon_block_slot=block.slot,
-        blobs=blobs,
-        kzg_aggregated_proof=compute_aggregate_kzg_proof(blobs),
-    )
+def get_blob_sidecars(block: BeaconBlock, blobs: Sequence[Blob]) -> Sequence[BlobSidecar]:
+    return [
+        BlobSidecar(
+            block_root=hash_tree_root(block),
+            index=index,
+            slot=block.slot,
+            block_parent_root=block.parent_root,
+            blob=blob,
+            kzg_commitment=block.body.blob_kzg_commitments[idx],
+            kzg_proof=compute_kzg_proof(blob),
+        )
+        for index, blob in enumerate(blobs)
+    ]
+
 ```
 
-This `signed_beacon_block_and_blobs_sidecar` is then published to the global `beacon_block_and_blobs_sidecar` topic.
+Then `signed_sidecar = SignedBlobSidecar(message=sidecar, signature=signature)` is constructed and to the global `blob_sidecar_{index}` topics according to its index.
+
+`signature` is obtained from:
+
+```python
+def get_blob_sidecar_signature(state: BeaconState,
```

Comment:
There is a proposal by @realbigsean in https://github.com/ethereum/beacon-APIs/issues/300 to only support blinded blob communication / signing in the validator client - not sure if it make sense to include it here once we have consensus on that proposal?

Reply:
I'm not entirely convinced it's a good idea yet - it does open up an imbalance which may come back to bite us later.

Reply:
I'm okay leaving the spec as signing over the full message even if in practice, APIs pass along a blinded or digest version.

I don't see a reason to creep this VC<>BN detail into the core specs